### ISSUE TITLE
Support of PHP 8.x

### ISSUE THREAD

**Issue by lyssar** [leafsphp/cli#1]
At the moment you can't install the leaf cli with php8.0. There are some constraint which are conflicting:
```php
> php -v
PHP 8.0.14 (cli) (built: Dec 20 2021 21:22:57) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.14, Copyright (c) Zend Technologies
    with Zend OPcache v8.0.14, Copyright (c), by Zend Technologies

> composer global require leafs/cli:^2.0
Changed current directory to /home/shens/.config/composer
./composer.json has been updated
Running composer update leafs/cli
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - leafs/cli[v2.0, ..., v2.0.2] require php ^7.2 -> your php version (8.0.14) does not satisfy that requirement.
    - leafs/cli[v2.0.3, ..., v2.0.4] require symfony/console ^5.0 -> found symfony/console[v5.0.0, ..., v5.4.3] but the package is fixed to v4.4.30 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires leafs/cli ^2.0 -> satisfiable by leafs/cli[v2.0, ..., v2.0.4].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

```

Active support ended 28 Nov 2021 | 2 months ago
And security updates will end on 28 Nov 2022 | in 9 month

Do you have any plans to upgrade to PHP8.0?

**Comment by mychidarko** [comment]
Please try running `composer global require leafs/cli` without the version constraints. This particular issue was fixed last year.

![image](https://user-images.githubusercontent.com/26604242/152259528-4a470f28-0caf-4982-b01c-6d062c94f9c8.png)

**Comment by lyssar** [comment]
:) worked for me. thnx

**Comment by bogordesaincom** [comment]
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - leafs/cli[v2.0, ..., v2.0.2] require php ^7.2 -> your php version (8.0.15) does not satisfy that requirement.
    - leafs/cli[v2.0.3, ..., v2.0.7] require symfony/process ^5.0 -> found symfony/process[v5.0.0, ..., v5.4.5] but the package is fixed to v6.0.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - leafs/cli v2.0.5 requires symfony/console ^6.0 -> found symfony/console[v6.0.0, ..., v6.0.5] but the package is fixed to v5.4.3 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.
    - Root composer.json requires leafs/cli ^2.0 -> satisfiable by leafs/cli[v2.0, ..., v2.0.7].

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions.
You can also try re-running composer require with an explicit version constraint, e.g. "composer require leafs/cli:*" to figure out if any version is installable, or "composer require leafs/cli:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.
```

**Comment by noraj** [comment]
Seems there is a regression, `composer global require leafs/cli` does not work anymore with php 8.x (in my case 8.2.7).

![image](https://github.com/leafsphp/cli/assets/16578570/e702248d-6e02-4c97-94b5-a4bf5345c167)
